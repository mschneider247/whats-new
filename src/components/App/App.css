* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.app {
  display: flex;
  background: #141527;
  height: 100vh;
  text-align: center;
  height: 100%;
  width: 100%;
  color: white;
}

.newsContainer {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr;
}
