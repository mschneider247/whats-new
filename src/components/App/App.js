import React, { Component } from 'react';
import entertainment from '../../data/entertainment';
import health from '../../data/health';
import local from '../../data/local';
import science from '../../data/science';
import technology from '../../data/technology';
import './App.css';
import SearchForm from '../SearchForm/SearchForm'
import NewsContainer from '../NewsContainer/NewsContainer'
import Menu from '../Menu/Menu'

const data = {
  default: [{id: 0, headline: 'Loading...', description: 'Please Wait...'}],
  entertainment: entertainment,
  health: health,
  local: local,
  science: science,
  technology: technology
}

// Should be able to fetch data and then update the state
// data.default works for now...

class App extends Component {
  constructor() {
    super();
    this.state = {
      news: data.default
    }
  }

  filterNews = (filter) => {
    this.setState({ news: data[filter]})
  }

  // search news...  needs to look at each key of data
  // then search that array?  hopefully
  // in the array we will check if any titles or descriptions
  // include a set of characters provided by the user
  // this will be the current state of our SearchForm class...

  render () {
    return (
      <main className="app">
        <section className="header">
          <SearchForm />
        </section>
        <section className="menu">
          <Menu filterNews={this.filterNews}/>
        </section>
        <section className="newsContainer">
          <NewsContainer news={this.state.news}/>
        </section>
      </main>
    );
  }
}

export default App;
