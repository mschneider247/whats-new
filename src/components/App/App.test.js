import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change state when filterNews method is called', () => {
    const initialState = wrapper.state();
    wrapper.instance().filterNews('technology');
    wrapper.instance().filterNews('default')
    expect(wrapper.state()).toEqual(initialState);
  })
  
})
