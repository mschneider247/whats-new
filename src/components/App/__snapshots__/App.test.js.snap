// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="app"
>
  <section
    className="header"
  >
    <SearchForm />
  </section>
  <section
    className="menu"
  >
    <Menu
      filterNews={[Function]}
    />
  </section>
  <section
    className="newsContainer"
  >
    <NewsContainer
      news={
        Array [
          Object {
            "description": "Please Wait...",
            "headline": "Loading...",
            "id": 0,
          },
        ]
      }
    />
  </section>
</main>
`;
